Source/HelixToolkit.Wpf.Shared/Helpers/Viewport3DHelper.cs

```diff
@@ -223,48 +223,65 @@ public static IEnumerable<RectangleHitResult> FindHits(this Viewport3D viewport,
                 (model, visual, transform) =>
                 {
                     var geometry = model.Geometry as MeshGeometry3D;
-                    if (geometry == null || geometry.Positions == null || geometry.TriangleIndices == null)
+                    if (geometry == null
+                    || model.Bounds.IsEmpty
+                    || geometry.Bounds.IsEmpty
+                    || geometry.Positions == null
+                    || geometry.Positions.Count == 0
+                    || geometry.TriangleIndices == null
+                    || geometry.TriangleIndices.Count == 0)
                     {
                         return;
                     }
+                    // ref https://learn.microsoft.com/en-us/archive/blogs/wpf3d/transforming-bounds
+                    GeneralTransform3DTo2D transform3DTo2D = visual.TransformToAncestor(viewport);
+                    // If null, the transform isn't possible at all
+                    if (transform3DTo2D is null) return;
+                    Rect visualBound = transform3DTo2D.TransformBounds(VisualTreeHelper.GetDescendantBounds(visual));
 
-                    var status = mode == SelectionHitMode.Inside;
-
-                    // transform the positions of the mesh to screen coordinates
-                    var point2Ds = geometry.Positions.Select(transform.Transform).Select(viewport.Point3DtoPoint2D).ToArray();
-
-                    // evaluate each triangle
-                    for (var i = 0; i < geometry.TriangleIndices.Count / 3; i++)
+                    if (rectangle.Contains(visualBound)) // object is both inside and touch
+                    {
+                        results.Add(new RectangleHitResult(model, visual));
+                    }
+                    else if (rectangle.IntersectsWith(visualBound))
                     {
-                        var triangle = new Triangle(
-                            point2Ds[geometry.TriangleIndices[i * 3]],
-                            point2Ds[geometry.TriangleIndices[(i * 3) + 1]],
-                            point2Ds[geometry.TriangleIndices[(i * 3) + 2]]);
-                        switch (mode)
+                        // transform the positions of the mesh to screen coordinates
+                        var point2Ds = geometry.Positions.Select(transform.Transform).Select(viewport.Point3DtoPoint2D).ToArray();
+                        var meshPoints = new Point[geometry.TriangleIndices.Count];
```

Comment:
I think you only need to transform all vertices instead of each triangles since multiple triangles may share same vertices.

Reply:
That's the problem from cutting plane, it does not remove the vertices after cutting.  And it should only affect the bounding box, not the final result from the selection, since it still does the intersection check with each triangles.

Reply:
No plans for fixing it. I only maintain sharpdx versions at this moment.